### PR TITLE
Fixing broken link for the Eq trait

### LINKED ISSUE #73233: Broken link in PartialEq documentation
In the documentation for [`PartialEq`][1], you will find the following:

> This trait allows for partial equality, for types that do not have a full equivalence relation. For example, in floating point numbers `NaN != NaN`, so floating point types implement `PartialEq` but not [`Eq`][2].

The link on this page is not actually a link to the `Eq` trait, but a link to the `PartialEq::eq` method on the same page.

The link is written as [\`Eq\`] with an [\`eq\`]: PartialEq::eq in the same doc-comment, so I am unsure whether this should be considered an issue in the documentation, or in rustdoc.

[1]: https://doc.rust-lang.org/stable/std/cmp/trait.PartialEq.html
[2]: https://doc.rust-lang.org/stable/std/cmp/trait.PartialEq.html#tymethod.eq

This issue has been assigned to @poliorcetics via [this comment](https://github.com/rust-lang/rust/issues/73233#issuecomment-643493334).

### PR DESCRIPTION
**Pull request by poliorcetics**
Fixes #73233.

### CODE REVIEW

File: src/libcore/cmp.rs [hunk 1]
```diff
@@ -35,7 +35,7 @@ use self::Ordering::*;
 ///
 /// This trait allows for partial equality, for types that do not have a full
 /// equivalence relation. For example, in floating point numbers `NaN != NaN`,
-/// so floating point types implement `PartialEq` but not [`Eq`].
+/// so floating point types implement `PartialEq` but not [`Eq`](Eq).
```
Comment (GuillaumeGomez) [inline src/libcore/cmp.rs:38]: Better leaving it like it was (``[`Eq`]``). Just removing the ``/// [`Eq`]: Eq`` line should be enough.

**Review by GuillaumeGomez — CHANGES_REQUESTED** [review]

File: src/libcore/cmp.rs (hunk 1 above)
Comment (poliorcetics) [reply src/libcore/cmp.rs:38]: Doing so makes the `` [`Eq`] `` link point to ``[`eq`]`` (which is a bug in `rustdoc` I think, but something we have to get around for now).

**Comment by JohnCSimon** [comment]
Ping from triage 
@poliorcetics  - please address the change request from GuillaumeGomez?
Thank you.

**Comment by poliorcetics** [comment]
The suggested change brings back the bug this PR fixes, except if rustdoc itself has been modified in between.

**Pull request merged**